include/libdnf/conf/option.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_CONF_OPTION_HPP
+#define LIBDNF_CONF_OPTION_HPP
+
+#include "libdnf/utils/exception.hpp"
+
+#include <string>
+
+namespace libdnf {
+
+class Option {
+public:
+    enum class Priority {
+        EMPTY = 0,
+        DEFAULT = 10,
+        MAINCONFIG = 20,
+        AUTOMATICCONFIG = 30,
+        REPOCONFIG = 40,
+        PLUGINDEFAULT = 50,
+        PLUGINCONFIG = 60,
+        DROPINCONFIG = 65,
```

Comment:
I think that priority for dropin config should be lower than 20. Please could you explain it in more details?
Also I am not sure what kind of problem we try to resolve by dropin configuration. Also explanation is required.
What I know that we cannot change a configuration during lifecycle of the system, because after modification of main config, it is not updated automatically by RPM. But dropin configuration will not resolve the issue.

Reply:
Drop-in is used to overwrite configuration files. So, it must have higher priority. Question is which configuration files may be overwritten by drop-in configuration?
On the other hand. I don't like the used priority set. It is copy from old libdnf. I do it from compatibility point of view. But I'm not sure in this case. Do you have an another set of priorities?